src/setup/node.rs

```diff
@@ -121,14 +147,22 @@ impl Node {
             .await
             .expect("Couldn't load the node's addresses.");
 
-        // TODO(Rqnsom) wait for the connection to confirm.
+        Node::wait_for_start(self.conf.net_addr.unwrap()).await;
     }
 
     /// Stops the node instance.
     pub fn stop(&mut self) -> io::Result<ChildExitCode> {
         // Cannot use 'mut self' due to the Drop impl.
 
+        // Remove address files since addresses may change if the node is restarted.
+        let remove_file = |file_name| match fs::remove_file(self.conf.path.join(file_name)) {
+            Ok(()) => (),
+            Err(e) if e.kind() == std::io::ErrorKind::NotFound => (),
+            Err(e) => panic!("Unexpected error: {:?}", e),
+        };
```

Comment:
Hmm, maybe we could make it shorter:
```
        let remove_file = |file_name| match fs::remove_file(self.conf.path.join(file_name)) {
            Err(e) if e.kind() != std::io::ErrorKind::NotFound => panic!("Unexpected error: {:?}", e),
            _ => (),
        };
```

Reply:
it could probably be made even shorter with `if matches!` :detective:

Reply:
This was my try here:
```
        let remove_file = |file_name| {
            if matches!(
            fs::remove_file(self.conf.path.join(file_name)),
            Err(e) if e.kind() != io::ErrorKind::NotFound)
            {
                panic!("Unexpected error: {:?}", e);
            }
        };
```
It ended up even longer 🤔